src/twister2/plugin.py

```diff
@@ -34,6 +35,32 @@ def pytest_collect_file(parent, path):
         return YamlFile.from_parent(parent, path=Path(path))
 
 
+def pytest_generate_tests(metafunc: pytest.Metafunc):
+    # generate parametrized tests for each selected platform for ordinary pytest tests
+    if 'specification' not in metafunc.fixturenames:
```

Comment:
Do we want to use 'specification' fixture directly for every pytest added?
If we found testspec.yaml in a folder with tests, we can just take all test scenarios from that folder.

Reply:
maybe `autouse=True` in fixture parameters can be usefull
```
@pytest.fixture(autouse=True)
def specification() -> None:
    pass

def pytest_generate_tests(metafunc):
    # check if testspec.yaml exists
    metafunc.parametrize("specification", ["one", "two"])

def test_foobar():
    pass
```

Reply:
I am not sure if `autouse=True` may cause some performance issue. It will be added to each test even those created from yaml files (can be thousand of tests if you run for all platforms). But we can change it later if we will see problems.

Reply:
After all we should not use `autouse=True`, because we don't want to apply this for all tests but only those which have specification yaml. Also `parametrized` require to have name of fixture in test function arguments.